### PR TITLE
spidermonkey_1_8_5: drop

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
It's ancient.

I honestly don't know enough about couchdb to judge if anyone still needs these old versions.
1.x at least seems to have some security issues, according to #99750

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -109,6 +109,8 @@ mapAliases ({
   conntrack_tools = conntrack-tools; # added 2018-05
   cool-old-term = cool-retro-term; # added 2015-01-31
   coprthr = throw "coprthr has been removed."; # added 2019-12-08
+  couchdb = throw "couchdb was removed from nixpkgs, use couchdb3 instead"; # added 2021-03-03
+  couchdb2 = throw "couchdb2 was removed from nixpkgs, use couchdb3 instead"; # added 2021-03-03
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:113]: You forgot the alias for spidermonkey_1_8_5.

Comment (ajs124) [reply pkgs/top-level/aliases.nix:113]: I didn't.

**Comment by drewrisinger** [comment]
Misspelled name of "spidermonkey" (missing "r") in title of commit 4f6fab89d24756e694f256a4a35060552cc50b6d

**Review by drewrisinger — CHANGES_REQUESTED** [review]
* [x] Diff LGTM minus lack of alias on dropping ``spidermonkey_1_8_5``. At the very least an explanation on why it doesn't need an alias should be added to the PR
* [x] Commits LGTM
* [x] Builds via ``nix-review`` on x86-64-linux

```
2 packages removed:
couchdb (†2.3.1) spidermonkey (†1.8.5)

[nix-shell:~/.cache/nixpkgs-review/pr-117094/nixpkgs]$ nix-build ./nixos/tests/couchdb.nix
...
(1.22 seconds)
couchdb3: must succeed: /nix/store/542yvgbyvm5il795axmpmns7zkgrg640-curl-jq-check-GET-_all_dbs.sh
couchdb3 #   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
couchdb3 #                                  Dload  Upload   Total   Spent    Left  Speed
couchdb3 #   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0[  195.395806] couchdb[532]: [notice] 2021-05-03T12:01:28.141358Z couchdb@127.0.0.1 <0.595.0> 475f48db2a 127.0.0.1:5984 127.0.0.1 testadmin GET /_all_dbs 200 ok 97
couchdb3 # 100     3    0     3    0     0     15      0 --:--:-- --:--:-- --:--:--    17
couchdb3 # 0
(1.08 seconds)
(198.90 seconds)
couchdb3 # [  195.909455] couchdb[532]: [notice] 2021-05-03T12:01:28.642327Z couchdb@127.0.0.1 <0.346.0> -------- chttpd_auth_cache changes listener died because the _users database does not exist. Create the database to silence this notice.
test script finished in 199.31s
cleaning up
killing couchdb3 (pid 7)
(0.00 seconds)
/nix/store/mmfv586g284q27mzynw8h9yj44vb2zkn-vm-test-run-couchdb
```

**Comment by ajs124** [comment]
I'm not sure I'll ever understand why people constantly want to insert (IMO useless) entries to aliases.nix, when we still don't have a plan to ever clean up that file, but there you go.

**Review by drewrisinger — APPROVED** [review]

**Pull request merged**